### ISSUE TITLE
LLVM does not understand that inequality is transitive (even with constraint-elimination)

### ISSUE THREAD

**Issue by jrmuizel** [llvm/llvm-project#53273]
GCC is able to eliminate the third assert in the following code:
```c
#include <assert.h>
void f(int a, int b, int c) {
    assert(a < b);
    assert(b < c);
    assert(a < c);
}
```

https://gcc.godbolt.org/z/s9a9YM6GK

However, if I change the type from `int` to `unsigned int` LLVM is able to figure it out.

**Comment by thibautvdv** [comment]
Looks like gcc cannot eliminate the third assert when using `unsigned int`.

https://gcc.godbolt.org/z/Ph3bqjzbq

**Comment by fhahn** [comment]
Initial signed support landed in 0a781d98fb0d

**Comment by jrmuizel** [comment]
This seems fixed now.
https://gcc.godbolt.org/z/s9a9YM6GK
